api.go

```diff
@@ -79,6 +80,16 @@ func (a *API) Exec(command string) (string, error) {
 	return msg.msg, msg.err
 }
 
+// ExecEval executes commands which do not return a trailing newline.
+func (a *API) ExecEval(command string) (string, error) {
+	s, err := a.Exec(fmt.Sprintf(`eval "$(%s)\n"`, command))
```

Comment:
Being picky again; `err` should be checked here, not at the end. Even though it's the same thing in this particular case, later on mistakes can happen if left like this.

Reply:
Sorry, I'm not entirely what you mean by checking here. Something like this?

```diff
diff --git a/api.go b/api.go
--- a/api.go
+++ b/api.go
@@ -83,6 +83,9 @@ func (a *API) Exec(command string) (string, error) {
 // ExecEval executes commands which do not return a trailing newline.
 func (a *API) ExecEval(command string) (string, error) {
        s, err := a.Exec(fmt.Sprintf(`eval "$(%s)\n"`, command))
+       if err != nil {
+               return "", err
+       }

        // when formatting with a newline, the response contains an extra trailing backslash
        s = strings.TrimSuffix(s, "\\")
```

Reply:
Yes exactly, and at the very end it can `return strings.TrimSuffix(s, "\\"), nil`.

Reply:
Done!